internal/querynodev2/segments/segment_loader.go

```diff
@@ -792,22 +793,36 @@ func (loader *segmentLoader) LoadDeltaLogs(ctx context.Context, segment Segment,
 	)
 	dCodec := storage.DeleteCodec{}
 	var blobs []*storage.Blob
+	var mtx sync.Mutex
+	var futures []*conc.Future[any]
 	for _, deltaLog := range deltaLogs {
 		for _, bLog := range deltaLog.GetBinlogs() {
+			bLog := bLog
 			// the segment has applied the delta logs, skip it
 			if bLog.GetTimestampTo() > 0 && // this field may be missed in legacy versions
 				bLog.GetTimestampTo() < segment.LastDeltaTimestamp() {
 				continue
 			}
-			value, err := loader.cm.Read(ctx, bLog.GetLogPath())
-			if err != nil {
-				return err
-			}
-			blob := &storage.Blob{
-				Key:   bLog.GetLogPath(),
-				Value: value,
-			}
-			blobs = append(blobs, blob)
+			future := GetLoadPool().Submit(func() (any, error) {
+				value, err := loader.cm.Read(ctx, bLog.GetLogPath())
+				if err != nil {
+					return nil, err
+				}
+				blob := &storage.Blob{
+					Key:   bLog.GetLogPath(),
+					Value: value,
+				}
+				mtx.Lock()
+				blobs = append(blobs, blob)
+				mtx.Unlock()
+				return nil, nil
```

Comment:
return the blob then you don't need the mutex

Reply:
yep, fixed.